src/idls/thrift/src/t_yarp_generator.cc

```diff
@@ -2218,6 +2239,48 @@ void t_yarp_generator::generate_struct_tostring(t_struct* tstruct, std::ostrings
     assert(indent_count_cpp() == 0);
 }
 
+void t_yarp_generator::generate_struct_typeconstexpr(t_struct* tstruct, std::ostringstream& f_h_, const std::string& yarp_type_name, const std::string& yarp_type_version)
+{
+    THRIFT_DEBUG_COMMENT(f_h_);
+
+    const auto& name = tstruct->get_name();
+
+    f_h_ << indent_h() << "//The name and the version for this message\n";
+    f_h_ << indent_h() << "static constexpr const char* typeName = \"" << yarp_type_name << "\";\n";
+    f_h_ << indent_h() << "static constexpr const char* typeVersion = \"" << "1.0" << "\";\n";
+    f_h_ << '\n';
+
+    assert(indent_count_h() == 1);
+}
+
+void t_yarp_generator::generate_struct_gettype(t_struct* tstruct, std::ostringstream& f_h_, std::ostringstream& f_cpp_)
+{
+    THRIFT_DEBUG_COMMENT(f_h_);
+    THRIFT_DEBUG_COMMENT(f_cpp_);
+
+    const auto& name = tstruct->get_name();
+
+    f_h_ << indent_h() << "// Get the message type\n";
+    f_h_ << indent_h() << "yarp::os::Type getType() const;\n";
+    f_h_ << '\n';
+
+    f_cpp_ << indent_cpp() << "// Convert to a printable string\n";
+    f_cpp_ << indent_cpp() << "yarp::os::Type " << name << "::getType() const\n";
+    f_cpp_ << indent_cpp() << "{\n";
+    indent_up_cpp();
+    {
+        f_cpp_ << indent_cpp() << " yarp::os::Type typ = yarp::os::Type::byNameOnWire(typeName);\n";
```

Comment:
`yarp::os::Type::byNameOnWire(typeName)` is equivalent to `yarp::os::Type::byName("yarp/bottle", typeName)`. We should definitely investigate what's the meaning of the first string, and if it is actually correct to use `byNameOnWire`, or it should be `yarp::os::Type::byName(typeName, typeName)`